### PR TITLE
[OASIS-20] Reenable most of the OTel components meant to be part of collector-contrib 

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

Reenable most of the OTel components that needed to be disabled when collector-contrib was added by https://github.com/DataDog/datadog-agent/pull/24836

### CODE REVIEW

**Review by songy23 — APPROVED** [review]
LGTM

**Review by truthbk — APPROVED** [review]
LGTM 👍

File: comp/otelcol/collector-contrib/impl/go.mod [hunk 1]
```diff
@@ -7,12 +7,14 @@ replace github.com/DataDog/datadog-agent/comp/otelcol/collector-contrib/def => .
...
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector v0.98.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector v0.98.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/exceptionsconnector v0.98.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/grafanacloudconnector v0.98.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/routingconnector v0.98.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector v0.100.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector v0.98.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/fileexporter v0.98.0
-	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter v0.98.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter v0.100.0
```
Comment (dineshg13) [inline comp/otelcol/collector-contrib/impl/go.mod:17]: should we update all of them to v0.100.0 ?

**Review by dineshg13 — APPROVED** [review]

File: comp/otelcol/collector-contrib/impl/go.mod (hunk 1 above)
Comment (dustmop) [reply comp/otelcol/collector-contrib/impl/go.mod:17]: Perhaps as a later change, right now I'm just trying to get the disabled components to be reenabled. Updating version numbers for everything requires an unpredictable amount of work, due to the need to get dependencies to work together, and licenses and copyrights to be correct, such that the linters are happy.

**Review by chouetz — APPROVED** [review]

**Review by jmeunier28 — APPROVED** [review]

File: .gitlab/binary_build/otel_agent.yml [hunk 2]
```diff
@@ -9,6 +9,9 @@
     - !reference [.retrieve_linux_go_deps]
     - inv -e otel-agent.build
   needs: ["go_deps"]
+  variables:
+    KUBERNETES_MEMORY_REQUEST: "8Gi"
+    KUBERNETES_MEMORY_LIMIT: "8Gi"
```
Comment (spencergilbert) [inline .gitlab/binary_build/otel_agent.yml:14]: Just noting this is only 2Gi higher than the default, may need more.

**Review by spencergilbert — APPROVED** [review]
Approved for BaRX owned file

File: .gitlab/binary_build/otel_agent.yml (hunk 2 above)
Comment (dustmop) [reply .gitlab/binary_build/otel_agent.yml:14]: Seems like this is at least getting the CI jobs to pass, but we'll keep an eye on it!

**Pull request merged**